packages/state_queue_test/lib/src/state_queue_test.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:async';
+import 'package:flutter/widgets.dart';
+import 'package:flutter/foundation.dart';
+import 'package:meta/meta.dart';
+import 'package:state_queue/state_queue.dart';
+import 'package:test/test.dart' as test;
+
+@isTest
+void blocTest<B extends StateQueue<State>, State>(
+  String description, {
+  @required B Function() build,
+  FutureOr<void> Function(B bloc) act,
+  Iterable<State> expect,
```

Comment:
```suggestion
  Matcher expect,
```

I would keep this more "open", it would still allow you to use `equals([...])` but also allow something like `isA<>` matcher.

Reply:
I think the difference between `Iterable` (which would be states or `Matcher`s) vs a single `Matcher` is important though.

My personal guess is that you mostly want state item level checks, and not so much globals (like `isEmpty`, `containsMoreThan(5)`, or similar).

So `Iterable<Matcher|State>` or a single `Matcher`?

Reply:
Just `Iterable` because that is what works in the end with the `expect` function. And we don't want to take away functionality, right?

Reply:
> Just `Iterable` because that is what works in the end with the `expect` function. And we don't want to take away functionality, right?

But then you can't use top-level matchers like `isEmpty`, and also need not use `equals([…])` as shown above.

Which is fine I guess, as I think asserting on the individual states is the focus for this kind of test.

Reply:
Yes, you are always going to get out an array/iterable to assert stuff on anyways. So if you want to test if it is empty you just put it an empty array.
And with only `Iterable` you can then either put in exact states to match or an array of Matchers to further testing on the states. This is very useful if the state is somewhat complex and you don't want to create it entirely for the test. Then you could just use an `isA` matcher and pick some properties to verify

Reply:
Allowed `Matcher`s inside the `Iterable` in 730e740095822c7c1a489bba736a5fb8925bbd45